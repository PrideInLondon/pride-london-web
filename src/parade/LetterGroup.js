import React from 'react'
import PropTypes from 'prop-types'

import { Heading, GroupsContainer, Wrapper } from './LetterGroup.styles'

const LetterGroup = ({ children, letter }) => {
  return (
    <Wrapper id={`parade-group-${letter}`}>
      <Heading>
        <h4>{letter.toUpperCase()}</h4>
      </Heading>
      <GroupsContainer>{children}</GroupsContainer>
    </Wrapper>
  )
}

LetterGroup.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
  letter: PropTypes.string.isRequired,
}

export default LetterGroup
