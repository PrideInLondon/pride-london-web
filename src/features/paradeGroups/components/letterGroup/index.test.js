import React from 'react'
import { shallow } from 'enzyme'
import LetterGroup from '.'

describe(LetterGroup.name, () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <LetterGroup letter="H">
        <p>Child</p>
      </LetterGroup>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
