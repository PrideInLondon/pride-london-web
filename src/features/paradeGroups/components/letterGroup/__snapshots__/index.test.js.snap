// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterGroup should render with default props 1`] = `
<styles__Wrapper
  id="parade-group-H"
>
  <styles__Heading>
    <h4>
      H
    </h4>
  </styles__Heading>
  <styles__GroupsContainer>
    <p>
      Child
    </p>
  </styles__GroupsContainer>
</styles__Wrapper>
`;
